File: Atcoder/mayo5/D.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int n, m;
    cin >> n >> m;
    vector<int> balls(n,0);
    for(int i = 0; i < n; ++i) cin >> balls[i];
    for(int i = 0; i < m; ++i){
        int t, hand;
        cin >> t;
        hand = balls[t];
        balls[t] = 0;
        int part = hand / n;
    }

    for(int i = 0; i < n; ++i) cout << balls[i] << " \n"[i==n-1];
    return 0;
}

File: Atcoder/mayo5/D.py
if __name__ == "__main__":
    n, m = map(int, input().split())
    balls = list(map(int, input().split()))
    for i in input().split():
        c = 0
        i = int(i)
        hand = balls[i]
        balls[i] = 0
        while hand:
            hand-=1
            c+=1
            balls[(i + c)%n]